src/miral/runner.cpp

```diff
@@ -134,8 +144,19 @@ try
         server->set_command_line(argc, argv);
         server->apply_settings();
         apply_env_hacks(*server);
-
+        
         weak_server = server;
+
+        auto const main_loop = server->the_main_loop();
+
+        for (auto const& signal : signal_backlog)
+        {
+            main_loop->register_signal_handler(signal.signals, signal.handler);
+        }
+        signal_backlog.clear();
+
+        fd_manager->set_weak_main_loop(std::move(main_loop));
```

Comment:
I'm not sure why this is working, it's generally not possible to move a const value (since moving implies changing the origin to null). Perhaps `main_loop` should not be const on line 150?